moveit_experimental/moveit_jog_arm/src/low_pass_filter.cpp

```diff
@@ -38,15 +38,44 @@
  */
 
 #include <moveit_jog_arm/low_pass_filter.h>
+#include <cmath>
 #include <string>
+#include <ros/ros.h>
 
-static const std::string LOGNAME = "low_pass_filter";
+const std::string LOGNAME = "low_pass_filter";
+constexpr double EPSILON = 1e-9;
 
 namespace moveit_jog_arm
 {
 LowPassFilter::LowPassFilter(double low_pass_filter_coeff)
+  : previous_measurements_{ 0., 0. }
+  , previous_filtered_measurement_(0.)
+  , filter_coeff_(low_pass_filter_coeff)
+  , scale_term_(1. / (1. + low_pass_filter_coeff))
+  , feedback_term_(1. - low_pass_filter_coeff)
 {
-  filter_coeff_ = low_pass_filter_coeff;
+  // guarantee this doesn't change because the logic depends on this length implicity
+  static_assert(LowPassFilter::FILTER_LENGTH == 2, "moveit_jog_arm::LowPassFilter::FILTER_LENGTH should be 2");
```

Comment:
Isn't that a little paranoid? Also, what if someone initializes `previous_measurements_` with another constant... ;)

Reply:
I don't think it is too paranoid.  The indexes of the array are explicitly used instead of loops below this and this check is just a compiler check.  It won't add any code to the binary.

I don't know why changing the values of `previous_measurements_` with the `reset` method would affect this?

Reply:
Sure, initialization using `reset` is not an issue. This check here only warns other developers from changing the value of `FILTER_LENGTH` in the header file. But you could just declare `previous_measurement_` with another constant and this assertion will never be triggered, so it's not really a guarantee that logic doesn't break. I'm not concerned about performance, I just think that a simple comment in the header file is more effective than letting developers run into warnings at compile time.